x-pack/filebeat/module/threatintel/malwarebazaar/ingest/pipeline.yml

```diff
@@ -0,0 +1,201 @@
+    target_field: threatintel.indicator.geo.country_iso_code
+    ignore_missing: true
+- rename:
+    field: threatintel.malwarebazaar.signature
+    target_field: threatintel.indicator.signature
+    ignore_missing: true
+- foreach:
+    field: threatintel.malwarebazaar.code_sign
+    ignore_missing: true
+    processor:
+      rename:
+        field: subject_cn
+        target_field: threatintel.indicator.file.x509.subject.common_name
+      rename:
+        field: issuer_cn
+        target_field: threatintel.indicator.file.x509.issuer.common_name
+      rename:
+        field: algorithm
+        target_field: threatintel.indicator.file.x509.public_key_algorithm
+      rename:
+        field: valid_from
+        target_field: threatintel.indicator.file.x509.not_before
+      rename:
+        field: valid_to
+        target_field: threatintel.indicator.file.x509.not_after
+      rename:
+        field: serial_number
+        target_field: threatintel.indicator.file.x509.serial_number
+- rename:
+    field: threatintel.malwarebazaar.file_size
+    target_field: threatintel.indicator.file.size
+    ignore_missing: true
+- rename:
+    field: threatintel.malwarebazaar.file_type
+    target_field: threatintel.indicator.file.extension
+    ignore_missing: true
+- rename:
+    field: threatintel.malwarebazaar.md5_hash
+    target_field: threatintel.indicator.file.hash.md5
+    ignore_missing: true
+- rename:
+    field: threatintel.malwarebazaar.sha256_hash
+    target_field: threatintel.indicator.file.hash.sha256
+    ignore_missing: true
+- rename:
+    field: threatintel.malwarebazaar.sha1_hash
+    target_field: threatintel.indicator.file.hash.sha1
+    ignore_missing: true
+- rename:
+    field: threatintel.malwarebazaar.sha3_384_hash
+    target_field: threatintel.indicator.file.hash.sha384
+    ignore_missing: true
+- rename:
+    field: threatintel.malwarebazaar.imphash
+    target_field: threatintel.indicator.file.pe.imphash
+    ignore_missing: true
+- rename:
+    field: threatintel.malwarebazaar.ssdeep
+    target_field: threatintel.indicator.file.hash.ssdeep
+    ignore_missing: true
+- rename:
+    field: threatintel.malwarebazaar.tlsh
+    target_field: threatintel.indicator.file.hash.tlsh
+    ignore_missing: true
+- append:
+    field: related.hash
+    value: '{{ threatintel.indicator.file.hash.md5 }}'
+    if: ctx?.threatintel?.indicator?.file?.hash?.md5 != null
+- append:
+    field: related.hash
+    value: '{{ threatintel.indicator.file.hash.sha256 }}'
+    if: ctx?.threatintel?.indicator?.file?.hash?.sha256 != null
+- append:
+    field: related.hash
+    value: '{{ threatintel.indicator.file.hash.ssdeep }}'
+    if: ctx?.threatintel?.indicator?.file?.hash?.ssdeep != null
+- append:
+    field: related.hash
+    value: '{{ threatintel.indicator.file.pe.imphash }}'
+    if: ctx?.threatintel?.indicator?.file?.pe?.imphash != null
+- append:
+    field: related.hash
+    value: '{{ threatintel.indicator.file.pe.tlsh }}'
```

Comment:
I'll see if maybe we can put this in as an official ECS field, but just wondering, why `pe.tlsh`? `tlsh` is a generic fuzzy hash, so it should pretty much be wherever `ssdeep` is.

Reply:
also, it doesn't look like it's in the pipeline, and I'm sure we could look into adding it in to the `elf` RFC, but what about `telfhash`? Thinking we could map it to `threatintel.indicator.file.elf.telfhash` and probably dump into `related.hash` as well.

Reply:
> I'll see if maybe we can put this in as an official ECS field, but just wondering, why `pe.tlsh`? `tlsh` is a generic fuzzy hash, so it should pretty much be wherever `ssdeep` is (i.e. `hash.tlsh`).

I see now. I copied the existing ingest pipeline from `abusemalware` and that appears to be where the issue is. `abusemalware` and `malwarebazaar` need to use `*.hash.tlsh` vs. `*.pe.tlsh`.
```
- append:
    field: related.hash
    value: '{{ threatintel.indicator.file.hash.tlsh }}'
    if: ctx?.threatintel?.indicator?.file?.hash?.tlsh != null
```

I'll make this correction and open an Issue for `abusemalware`.

Reply:
> also, it doesn't look like it's in the pipeline or filled in in your source documents (just looks like it's `null` everywhere), and I'm sure we could look into adding it in to the `elf` RFC, but what about `telfhash`? Thinking we could map it to `threatintel.indicator.file.elf.telfhash` and probably dump into `related.hash` as well.

I added `*.elf.telfhash`, good addition.

Reply:
`tlsh` bug for Abuse Malware
https://github.com/elastic/beats/issues/25151